### PR TITLE
[msbuild] Unify the CompileAppManifest target.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Also rework the class hierarchy a little bit, so that Xamarin.iOS and
Xamarin.Mac are identical:

    CompileAppManifestTaskBase
    └─── iOS/CompileAppManifestTaskCore
    │    └─── iOS/CompileAppManifest
    └─── Mac/CompileAppManifestTaskCore
        │    └─── Mac/CompileAppManifest

### CODE REVIEW

File: msbuild/Xamarin.MacDev.Tasks.Core/PlatformFramework.cs [hunk 1]
```diff
@@ -66,5 +66,27 @@ public static string GetMinimumOSVersionKey (ApplePlatform platform)
 				throw new InvalidOperationException ($"Invalid platform: {platform}");
 			}
 		}
+
+		public static string GetMinimumVersionOperatingSystem (string targetFrameworkMoniker, bool is_simulator)
```
Comment (spouliot) [inline msbuild/Xamarin.MacDev.Tasks.Core/PlatformFramework.cs:70]: method name mention version but it only returns platform/OS names, no version information

File: msbuild/Xamarin.Mac.Tasks/Tasks/CompileAppManifest.cs [hunk 2]
```diff
@@ -6,50 +6,7 @@
...
-			// Merge with any partial plists generated by the Asset Catalog compiler...
-			MergePartialPlistTemplates (plist);
-
-			CompiledAppManifest = new TaskItem (Path.Combine (AppBundleDir, "Contents", "Info.plist"));
-			plist.Save (CompiledAppManifest.ItemSpec, true, true);
-
-			return !Log.HasLoggedErrors;
-		}
+namespace Xamarin.Mac.Tasks {
+	public class CompileAppManifest : CompileAppManifestTaskCore {
```
Comment (spouliot) [inline msbuild/Xamarin.Mac.Tasks/Tasks/CompileAppManifest.cs:10]: semi-unrelated, but it's a recurring issue fixed here
we could use the cecil-based tests to ensure `Core` tasks have the code and the non-`Core` code are empty

File: msbuild/Xamarin.MacDev.Tasks.Core/PlatformFramework.cs (hunk 1 above)
Comment (rolfbjarne) [reply msbuild/Xamarin.MacDev.Tasks.Core/PlatformFramework.cs:70]: It was supposed to be something like `GetOperatingSystemAsPassedToMinimumVersionString`, but I found that a little bit too long. There's already a `GetOperatingSystem` method in this class that returns different values, so I had to come up with something different. I've added a comment to explain the usage.

Comment (rolfbjarne) [reply msbuild/Xamarin.MacDev.Tasks.Core/PlatformFramework.cs:71]: Actually these methods are unrelated to this PR, so I just removed them, they're only in #8875 now.

**Comment by rolfbjarne** [comment]
Test failures are unrelated

* */watchOS 32-bit: https://github.com/xamarin/maccore/issues/581
* xammac tests: will be fixed with #8886.

File: msbuild/Xamarin.Mac.Tasks/Tasks/CompileAppManifest.cs (hunk 2 above)
Comment (rolfbjarne) [reply msbuild/Xamarin.Mac.Tasks/Tasks/CompileAppManifest.cs:10]: > semi-unrelated, but it's a recurring issue fixed here
> we could use the cecil-based tests to ensure `Core` tasks have the code and the non-`Core` code are empty

@spouliot see https://github.com/xamarin/xamarin-macios/pull/8888

**Pull request merged**